### DISCUSSION TITLE
ModuleNotFoundError: No module named 'mxnet.gluon.metric'

### DISCUSSION THREAD

**Discussion by Johnny-dai-git** [apache/incubator-mxnet#19341] [Q&A] [answered]
Hi, While I am trying to run the example from mxnet/example/gluon.

I meet this problem:

root@server:/users/yxd429/mxnet2/mxnet/example/gluon# /usr/bin/python3 image_classification.py --dataset cifar10 --model vgg11 --epochs 1
Traceback (most recent call last):
  File "image_classification.py", line 30, in <module>
    from mxnet.gluon.metric import Accuracy, TopKAccuracy, CompositeEvalMetric
ModuleNotFoundError: No module named 'mxnet.gluon.metric'

I am trying to run the training no cpu. I install it as "pip3 install mxnet"

What is the problem should be ?

Best Regards,
Johnny

**Comment by leezu** [comment, marked as answer]
`pip3 install mxnet` installs MXNet 1 and you are executing MXNet 2 Python code with MXNet 1, which is expected to fail.

MXNet 2 is currently unreleased. You can track the status for beta release at https://github.com/apache/incubator-mxnet/issues/19139

You can use the v1.x branch to get MXNet 1 or compile MXNet 2 from source. We'll provide package on pypi once the beta is approved later. If you wish to help test MXNet 2, you can also install a nightly build `pip install --pre -f https://dist.mxnet.io/python/cpu mxnet`

**Comment by Johnny-dai-git** [comment]
What i did is i change the source code from mxnet.gluon.metrics to
mxnet.metrics.

It works now.

thank you

Leonard Lausen <notifications@github.com>于2020年10月12日 周一下午7:15写道：

> pip3 install mxnet installs MXNet 1 and you are executing MXNet 2 Python
> code with MXNet 1, which is expected to fail.
>
> MXNet 2 is currently unreleased. You can track the status for beta release
> at #19139 <https://github.com/apache/incubator-mxnet/issues/19139>
>
> You can use the v1.x branch to get MXNet 1 or compile MXNet 2 from source.
> We'll provide package on pypi once the beta is approved later. If you wish
> to help test MXNet 2, you can also install a nightly build pip install
> --pre -f https://dist.mxnet.io/python/cpu mxnet
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-mxnet/discussions/19341#discussioncomment-95694>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDGLBYSPPHWCTE4ZYV3MJDSKOE2FANCNFSM4SNTKD5Q>
> .
>
-- 
Yuanjun Dai (he/him)
P.hd
Department of Computer and Data Sciences
Case Western Reserve University
Phone: (216)-235-8330
Office: Glennan 505

**Comment by Johnny-dai-git** [comment]
Hi, Leonard

Actually, I follow the instructions from the tutorial from this link.

I just try to run the example model here.

https://mxnet.apache.org/versions/1.7/api/faq/distributed_training

All the command I am using is from this link...

However, the errors keep showing up.

What tutorial I need to follow with ?

[apache/incubator-mxnet] ModuleNotFoundError: No module named
'mxnet.gluon.metric'

Best Regards,
Johnny



On Mon, Oct 12, 2020 at 7:15 PM Leonard Lausen <notifications@github.com>
wrote:

> pip3 install mxnet installs MXNet 1 and you are executing MXNet 2 Python
> code with MXNet 1, which is expected to fail.
>
> MXNet 2 is currently unreleased. You can track the status for beta release
> at #19139 <https://github.com/apache/incubator-mxnet/issues/19139>
>
> You can use the v1.x branch to get MXNet 1 or compile MXNet 2 from source.
> We'll provide package on pypi once the beta is approved later. If you wish
> to help test MXNet 2, you can also install a nightly build pip install
> --pre -f https://dist.mxnet.io/python/cpu mxnet
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-mxnet/discussions/19341#discussioncomment-95694>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDGLBYSPPHWCTE4ZYV3MJDSKOE2FANCNFSM4SNTKD5Q>
> .
>


-- 
Yuanjun Dai (he/him)
P.hd
Department of Computer and Data Sciences
Case Western Reserve University
Phone: (216)-235-8330
Office: Glennan 505

**Reply by szha** [reply]
The problem with the page is that it links to master branch, which actually should have been linked to a commit instead. i.e.

This page:
https://github.com/apache/incubator-mxnet/blame/v1.7.x/docs/static_site/src/pages/api/faq/distributed_training.md#L98
should link to:
https://github.com/apache/incubator-mxnet/blob/7efb5df7fcaaff5b1ced41b7e718e2146ed57814/example/gluon/image_classification.py

For now you can run the script from the second link

**Comment by Johnny-dai-git** [comment]
Thank you so much.


I will try.

Best Regards,
Johnny

On Sat, Oct 17, 2020 at 11:10 AM Sheng Zha <notifications@github.com> wrote:

> The problem with the page is that it links to master branch, which
> actually should have been linked to a commit instead. i.e.
>
> This page:
>
> https://github.com/apache/incubator-mxnet/blame/v1.7.x/docs/static_site/src/pages/api/faq/distributed_training.md#L98
> should link to:
>
> https://github.com/apache/incubator-mxnet/blob/7efb5df7fcaaff5b1ced41b7e718e2146ed57814/example/gluon/image_classification.py
>
> For now you can run the script from the second link
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-mxnet/discussions/19341#discussioncomment-102693>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDGLB7OHJUV6PWMDYKDC2LSLGXVHANCNFSM4SNTKD5Q>
> .
>


-- 
Yuanjun Dai (he/him)
P.hd
Department of Computer and Data Sciences
Case Western Reserve University
Phone: (216)-235-8330
Office: Glennan 505

**Comment by Johnny-dai-git** [comment]
Hi,

Thank you for your guys' quick response !

Hope this E-mail finds you well!

I use a right image_classification.py file, however, I still have other
problems.

Problem One:

Within the tutorial for distributed training, the information within the
hostfile should be only IP address.

My master can ssh to the worker nodes using command "ssh -A user@ip_address".
However, if the hostfile only contains the ip_address without the user
name, the ssh will be denied because of the publickkey problem. However, if
the hostfile contains user@ip_address, there will be no ssh problem.

Problem Two:

When I try to run the command:

../../tools/launch.py -n 3 -s 1 -H hosts --launcher ssh --sync-dst-dir
/users/yxd429/test python image_classification.py --dataset cifar10 --model
vgg11 --epochs 1 --kvstore dist_sync

There will be a errors:

rsync: failed to set times on "/users/yxd429/test/.": Operation not
permitted (1)


How can I solve this problem ?


Best Regards,

Johnny


-- 
Yuanjun Dai (he/him)
P.hd
Department of Computer and Data Sciences
Case Western Reserve University
Phone: (216)-235-8330
Office: Glennan 505

**Comment by Johnny-dai-git** [comment]
Also, even though I follow all the instructions, I still  got this type of
errors:

What is wrong with my implementation ?

root@server:/users/yxd429/incubator-mxnet/example/gluon#
../../tools/launch.py -n 3 -s 1 -H hosts --launcher ssh --sync-dst-dir
/users/yxd429/test python image_classification.py --dataset cifar10 --model
vgg11 --epochs 1 --kvstore dist_sync

2020-10-17 13:49:04,653 INFO rsync
/users/yxd429/incubator-mxnet/example/gluon/ -> yxd429@128.105.145.73
yxd429@128.105.145.69 yxd429@128.105.145.42 yxd429@128.105.145.221:
/users/yxd429/test

bash: yxd429@128.105.145.69: command not found

bash: line 0: export: `yxd429@128.105.145.69': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.42': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.221': not a valid identifier

python: can't open file 'image_classification.py': [Errno 2] No such file
or directory

bash: yxd429@128.105.145.69: command not found

bash: line 0: export: `yxd429@128.105.145.69': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.42': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.221': not a valid identifier

bash: yxd429@128.105.145.69: command not found

bash: yxd429@128.105.145.69: command not found

bash: line 0: export: `yxd429@128.105.145.69': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.42': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.221': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.69': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.42': not a valid identifier

bash: line 0: export: `yxd429@128.105.145.221': not a valid identifier


Best Regards,

Johnny



On Sat, Oct 17, 2020 at 11:10 AM Sheng Zha <notifications@github.com> wrote:

> The problem with the page is that it links to master branch, which
> actually should have been linked to a commit instead. i.e.
>
> This page:
>
> https://github.com/apache/incubator-mxnet/blame/v1.7.x/docs/static_site/src/pages/api/faq/distributed_training.md#L98
> should link to:
>
> https://github.com/apache/incubator-mxnet/blob/7efb5df7fcaaff5b1ced41b7e718e2146ed57814/example/gluon/image_classification.py
>
> For now you can run the script from the second link
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-mxnet/discussions/19341#discussioncomment-102693>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANDGLB7OHJUV6PWMDYKDC2LSLGXVHANCNFSM4SNTKD5Q>
> .
>


-- 
Yuanjun Dai (he/him)
P.hd
Department of Computer and Data Sciences
Case Western Reserve University
Phone: (216)-235-8330
Office: Glennan 505
